### PR TITLE
Update Workflow access request approver to respect RBAC

### PR DESCRIPTION
**Pull request by yjperez**
Removed the user allow section, modified the client to use SubmitAccessReview instead of SetAccessRequestState to mitigate rouge approval of access request.

### CODE REVIEW

**Comment by yjperez** [comment]
With the previous iteration of this plugin, it would approve allowed users for any role they requested. This modification now only approves roles that the access-plugin is given permission to review.

File: examples/workflows/main.go [hunk 1]
```diff
@@ -115,15 +115,19 @@ func run(ctx context.Context, cfg *config) error {
...
+					State:    types.RequestState_APPROVED,
+				    Decision: types.Decision{
+				        Allow: allowed,
+				        Details: &types.DecisionDetails{
+				            Reason: fmt.Sprintf("%s:%s", cfg.PluginName, delegator),
+				        },
+				    },
 				}
-
+
```
Comment (zmb3) [inline examples/workflows/main.go:130]: Remove this whitespace

File: examples/workflows/config.toml [hunk 2]
```diff
@@ -6,9 +6,3 @@ addr = "localhost:3025"
 
 # Teleport identity file with TLS/SSH certificates.
 identity_file = "access-plugin-identity"
-
-# allow_list determines which users' requests will be approved.
-allow_list = [
```
Comment (zmb3) [inline examples/workflows/config.toml:11]: Why was this removed? The code still looks at the allow list.

File: examples/workflows/main.go [hunk 3]
```diff
@@ -16,9 +16,9 @@ limitations under the License.
 
 // Example Allow List based access plugin.
 //
-// This plugin approves/denies access requests based on a simple Allow List
-// of usernames and associated allowed roles. Requests from allowed users
-// for allowed roles are approved, all others are denied.
+// This plugin atomatically approves access requests based on the approver's role.
```
Comment (zmb3) [inline examples/workflows/main.go:19]:
```suggestion
// This plugin automatically approves access requests based on the approver's role.
```

But it's also based on the allow list, no?

File: examples/workflows/main.go [hunk 4]
```diff
@@ -115,15 +115,19 @@ func run(ctx context.Context, cfg *config) error {
 				if !ok {
 					return trace.BadParameter("unexpected resource type %T", event.Resource)
 				}
-
+
```
Comment (zmb3) [inline examples/workflows/main.go:118]: Remove

File: examples/workflows/main.go (hunk 1 above)
Comment (yjperez) [reply examples/workflows/main.go]: Removed

File: examples/workflows/main.go [hunk 5]
```diff
@@ -114,40 +114,23 @@ func run(ctx context.Context, cfg *config) error {
...
-					params.Reason = "user in Allow List"
-				} else {
-					log.Printf("User %q not in Allow List, denying request...", req.GetUser())
-					params.State = types.RequestState_DENIED
-					params.Reason = "user not in Allow List"
-				}
-
+				params := types.AccessReviewSubmission{
+				    Resource: req.GetName(),
+					State:    types.RequestState_APPROVED,
```
Comment (zmb3) [inline examples/workflows/main.go:121]: This code is not properly formatted.

https://go.dev/blog/gofmt

File: examples/workflows/main.go [hunk 6]
```diff
@@ -16,9 +16,9 @@ limitations under the License.
 
 // Example Allow List based access plugin.
```
Comment (zmb3) [inline examples/workflows/main.go:17]: This (as well as the readme) still mentions an allow list, but you've removed all traces of the allow list.

Documentation that doesn't match the code creates additional support burden.

What is your intent here? That the plugin will attempt to automatically approve every single access request it sees?

**Review by zmb3 — CHANGES_REQUESTED** [review]

File: examples/workflows/main.go (hunk 6 above)
Comment (yjperez) [reply examples/workflows/main.go:17]: The allow list would attempt to approve all requests for users in the allow list regardless if the plugin role had permission or not. Removing the allow list so the plugin now will only approve requests for roles the plugin is authorized to review based on RBAC.

Will go over and remove all traces of allow list.

File: examples/workflows/main.go (hunk 5 above)
Comment (yjperez) [reply examples/workflows/main.go:115]: Formatted the code.

**Review by yjperez — COMMENTED** [review]
Made requested changes

**Comment by Joerger** [comment]
A couple questions / comments about the nature of the PR.

1. What is the reason for removing the Allow List and instead approving all requests that come in? Wouldn't this be akin to giving the roles to all users? 

2. This plugin uses the old access request flow, using `SetAccessRequestState`. This is the same flow used by the Jira Plugin currently, and is compatible with OSS. Maybe instead of removing the old example, we should just add a new example using `SubmitAccessReview`?

3. This plugin is purely intended as an example and should not be used without changes in the first place. For example, users could change it to an `AllowedRolesList` to get the functionality provided by this change. If users are getting confused and just using it verbatim, maybe we should remove the example.

File: examples/workflows/main.go (hunk 6 above)
Comment (zmb3) [reply examples/workflows/main.go]: I guess the concern is you're basically removing any notion of telling the plugin what to approve, and just telling to to try to approve everything it can, and letting RBAC catch errors. 

Will this clutter the audit log with access denied errors?

**Comment by zmb3** [comment]
Marking this one inactive. 

@yjperez please let us know if you plan on taking this to completion. If I don't hear from you I will assume this is no longer needed and close it out.

**Comment by Joerger** [comment]
Closing in favor of https://github.com/gravitational/teleport/pull/26850
